src/uct/base/uct_cm.c

```diff
@@ -51,6 +51,67 @@ ucs_status_t uct_cm_config_read(uct_component_h component,
     return UCS_OK;
 }
 
+ucs_status_t uct_cm_check_ep_params(const uct_ep_params_t *params)
+{
+    if (!(params->field_mask & UCT_EP_PARAM_FIELD_CM)) {
+        ucs_error("UCT_EP_PARAM_FIELD_CM is not set. field_mask 0x%lx",
+                  params->field_mask);
+        return UCS_ERR_INVALID_PARAM;
+    }
+
+    if (!(params->field_mask & UCT_EP_PARAM_FIELD_SOCKADDR_CB_FLAGS) ||
+        !(params->sockaddr_cb_flags & UCT_CB_FLAG_ASYNC)) {
+        ucs_error("UCT_EP_PARAM_FIELD_SOCKADDR_CB_FLAGS and UCT_CB_FLAG_ASYNC "
+                  "should be set. field_mask 0x%lx, sockaddr_cb_flags 0x%x",
+                  params->field_mask, params->sockaddr_cb_flags);
+        return UCS_ERR_UNSUPPORTED;
+    }
+
+    if (!(params->field_mask & (UCT_EP_PARAM_FIELD_SOCKADDR |
+                                UCT_EP_PARAM_FIELD_CONN_REQUEST))) {
+        ucs_error("neither UCT_EP_PARAM_FIELD_SOCKADDR nor "
+                  "UCT_EP_PARAM_FIELD_CONN_REQUEST is set. field_mask 0x%lx",
+                  params->field_mask);
+        return UCS_ERR_INVALID_PARAM;
+    }
+
+    return UCS_OK;
+}
+
+
+UCS_CLASS_INIT_FUNC(uct_cm_base_ep_t, const uct_ep_params_t *params)
+{
+    ucs_status_t status;
+
+    UCS_CLASS_CALL_SUPER_INIT(uct_base_ep_t, &params->cm->iface);
+
+    status = uct_cm_check_ep_params(params);
+    if (status != UCS_OK) {
+        return status;
+    }
+
+    self->priv_pack_cb      = (params->field_mask &
+                               UCT_EP_PARAM_FIELD_SOCKADDR_PACK_CB) ?
+                              params->sockaddr_pack_cb : NULL;
+    self->disconnect_cb     = (params->field_mask &
+                               UCT_EP_PARAM_FIELD_SOCKADDR_DISCONNECT_CB) ?
+                              params->disconnect_cb : NULL;
+    self->user_data         = (params->field_mask &
+                               UCT_EP_PARAM_FIELD_USER_DATA) ?
+                              params->user_data : NULL;
+
+    self->client.connect_cb = params->sockaddr_connect_cb.client;
```

Comment:
why client part is initialized unconditionally in common base code?
I would expect smth like 
```
if (client) {
    self->client.connect_cb = params->sockaddr_connect_cb.client;
} esle {
    self->server.connect_cb = params->sockaddr_connect_cb.server;
}
```